nodebuilder/state/core_accessor_stub.go

```diff
@@ -0,0 +1,116 @@
+package state
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+var ErrNoStateAccess = errors.New("node is running without state access")
+
+// stubbedCoreAccessor provides a stub for the state module to return
+// errors when state endpoints are accessed without a running connection
+// to a core endpoint.
+type stubbedCoreAccessor struct{}
```

Comment:
Let's call it after module. It's first of all the stub for the module, so that users running the node without state received a nice error if access the moduler through RPC.